Pidaq/Lib/uploadToDrive/uploadToDrive.py

```diff
@@ -0,0 +1,125 @@
+            and only a specific set of values are allowed.
+    The parent folder is supposedly going to be fixed for a
+    long period of time.
+    """
+    # searches for the parent name in the dictionary
+    for key in parentDictionary.keys():
+        if key.lower() == parentName.lower():
+            parentKey = key
+    try:
+        parentFolderID = parentDictionary[parentKey]
+    except KeyError:
+        # uploads to a default folder if the folder name does
+        # not exist in the dictionary
+        parentFolderID = DefaultID
+        logging.error('Folder name not found \nuploading to "Default"')
+    
+    #Today's date in YYYY-MM-DD format is stored in the variable "date"
+    date = datetime.today().strftime('%Y-%m-%d')
+    
+    # If the folder does not exist, should be the case
+    # when running for the first time on a given day
+    if checkExistingSubFolder(date, parentFolderID) == 0:
+        folder_metadata = {
+            'name' : date,
+            'mimeType': 'application/vnd.google-apps.folder',
+            'parents' : [parentFolderID]
+            }
+        # Creates a folder with the given date
+        folder = DRIVE.files().create(body = folder_metadata,
+                                      fields = 'id').execute()
+        # Returns the date folder's ID
+        folderId = getFolderId(date, parentFolderID)
+    # executes if the folder already exists  
+    elif checkExistingSubFolder(date, parentFolderID) == 1:
+        logging.warning("found folder for today's date")
+        folderId = getFolderId(date, parentFolderID)
+    # executes if more than one folder exists
+    else:
+        logging.warning('unexpected count of folders named with a given date')
+        return
+    
+    # extracts the .csv file name from the path
+    fileName=os.path.basename(filePath)
+    metadata = {'name': fileName,
+                'parents' : [folderId]
+                }
+    # uploads the .csv file
+    response = DRIVE.files().create(body=metadata,
+                               media_body=filePath).execute()
+    
+    if response:
+        logging.warning('Uploaded "%s" (%s)' % (metadata['name'], response['mimeType']))
+    else:
+        logging.warning('error in uploading the csv file')
+        
+def checkExistingSubFolder(folderName, parentFolder):
+    """ Queries the drive for the folder with
+        the given name and parent folder
+    """
+    query = " name = '" + folderName + "' and parents = \"" + parentFolder +'"'
+    response = DRIVE.files().list(q = query).execute()
+
+    # Returns 0 if no folder exists with today's date
+    # in the specified parent folder
+    if len(response.get('files',[])) == 0: 
+        logging.warning("creating a new folder...")
+        return 0
+    # Returns 1 if the folder with today's date
+    # exists in the specified parent folder
+    elif len(response.get('files',[])) == 1: 
+        for data in response.get('files', []):

```

Comment:
What does this line do? Couldn't you just return 1 if that condition evaluates to True? The for loop doesn't seem to be doing anything.